### PR TITLE
Release Candidate v0.0.8

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Bugfix release candidate to address changes to the Kp files.  Also includes other maintenance improvements to the F10.7 and Kp data.

# Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Download and load data for all Kp and F10.7 data sets.

## Test Configuration
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant: using develop branch of pysat

# Checklist:

- [x] Update the version number
- [x] Verify and update Zenodo information in .zenodo.json
- [x] doc tests pass locally
- [x] activate this branch on readthedocs (see CI below)
- [x] Upload the test distribution to the testPyPi server
- [x] Contacted developers and users about the future release
- [x] Performed integration tests with downstream packages (include list of these packages)
- [x] Create a wiki page with the [release text](https://github.com/pysat/pysatSpaceWeather/wiki/v0.0.8)
- [x] After approval, confirm all changes from the RC pull are listed in the wiki release text.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

**Review by jklenzing — COMMENTED** [review]
setup.cfg needs to include python 3.10

I don't see this branch activated on readthedocs.  Can't access anything through the github test

**Comment by aburrell** [comment]
> Can't access anything through the github test

I just clicked on "details" on the readthedocs CI and was directed to the docs for this branch.  Let me know if you continue to have issues.

**Review by jklenzing — COMMENTED** [review]
The code looks good, but some of the examples in the docs are out of date since instrument behaviour has changed.  I can either open a pull with suggestions or an issue to fix in the next version.

**Comment by aburrell** [comment]
Issue for the next version will do, thanks!

**Comment by aburrell** [comment]
Disregard, this should be fixed.  Can you give me a list of the examples?

**Comment by jklenzing** [comment]
In ex_init.rst:
```
import pysat
import pysatSpaceWeather as py_sw

f107 = pysat.Instrument(inst_module=py_sw.instruments.sw_f107,
                        tag='historic', update_files=True)
f107.download(start=f107.lasp_stime, stop=f107.today())
f107.load()
print(f107)
```
last line should be `print(f107.data)`, otherwise instrument info is displayed (tag, inst_id, etc) instead of data.

Output now has three columns:
```
             f107  f107_adjusted  f107_observed
1947-02-14  253.9          253.9            NaN
1947-02-17  228.5          228.5            NaN
1947-02-19  178.8          178.8            NaN
1947-02-20  163.7          163.7            NaN
1947-02-24  164.1          164.1            NaN
...           ...            ...            ...
2018-04-26   70.3           70.3           69.4
2018-04-27   69.6           69.6           68.7
2018-04-28   71.1           71.1           70.2
2018-04-29   72.2           72.2           71.1
2018-04-30   71.3           71.3           70.2

[25367 rows x 3 columns]
```

**Comment by jklenzing** [comment]
Not sure what is going on with the kp data, but here is my resulting plot:

kp_problem

**Review by jklenzing — APPROVED** [review]
Confirmed all examples in the docs are running as expected.

**Review by rstoneback — COMMENTED** [review]
Thanks @aburrell. Read through the code and found a few small areas with a question or concern.

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 1]
```diff
@@ -237,6 +237,10 @@ def load(fnames, tag='', inst_id=''):
                 meta.labels.desc: meta['f107_adjusted', meta.labels.desc]}
 
     elif tag == 'daily' or tag == 'prelim':
+        # Update the allowed types for the fill value
+        meta.labels.label_type['fill_val'] = (float, int, np.float64,
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysatSpaceWeather/instruments/sw_f107.py:241]: This is instrument specific and only affects the fill value for this instrument.  No other types are needed here.

File: pysatSpaceWeather/instruments/sw_f107.py [hunk 2]
```diff
@@ -486,30 +488,37 @@ def download(date_array, tag, inst_id, data_path, update_files=False):
...
-                raw_dict = json.loads(req.text)['noaa_radio_flux']
-                data = pds.DataFrame.from_dict(raw_dict['samples'])
-                if data.empty:
-                    warnings.warn("no data for {:}".format(dl_date),
-                                  UserWarning)
+                if req.ok:
+                    raw_dict = json.loads(req.text)['noaa_radio_flux']
+                    data = pds.DataFrame.from_dict(raw_dict['samples'])
+                    if data.empty:
+                        warnings.warn("no data for {:}".format(dl_date),
```
[the remark this answers is not in the document]
Comment (aburrell) [reply pysatSpaceWeather/instruments/sw_f107.py:495]: That would be a change in behaviour, so if you want that to happen it needs to be an issue that we discuss.

[the remark this answers is not in the document]
Comment (rstoneback) [reply pysatSpaceWeather/instruments/sw_f107.py:495]: In line 517 the logger is used to report no data is downloaded though the report is coming from a different part of the download code. I don't think it is a change in behavior. Up to you, I've reported my comment.

**Review by rstoneback — APPROVED** [review]

**Pull request merged**